### PR TITLE
Apply regex on PaintAddImageAsChildRotated

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is the third regex from https://github.com/OpenRCT2/OpenRCT2/pull/15253 and Closes https://github.com/OpenRCT2/OpenRCT2/issues/15288

based on #16776 first

### CODE REVIEW

File: src/openrct2/ride/gentle/SpiralSlide.cpp [hunk 1]
```diff
@@ -173,8 +173,8 @@ static void spiral_slide_paint_tile_front(
             image_id = (offset + slide_progress) | (ride.slide_peep_t_shirt_colour << 19) | (1 << 29);
 
             PaintAddImageAsChild(
-                session, image_id, 16, 16, boundingBox.x, boundingBox.y, boundingBox.z, height, boundingBoxOffset.x,
-                boundingBoxOffset.y, boundingBoxOffset.z);
+                session, image_id, { 16, 16, height }, { boundingBox.x, boundingBox.y, boundingBox.z },
+                { boundingBoxOffset.x, boundingBoxOffset.y, boundingBoxOffset.z });
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/SpiralSlide.cpp:177]:
```suggestion
                session, image_id, { 16, 16, height }, boundingBox, boundingBoxOffset);
```

File: src/openrct2/ride/transport/MiniatureRailway.cpp [hunk 2]
```diff
@@ -1145,8 +1145,8 @@ static void paint_miniature_railway_track_s_bend_left(
 
         imageId = miniature_railway_track_pieces_s_bend_left[direction & 1][trackSequence] | session.TrackColours[SCHEME_TRACK];
         PaintAddImageAsChildRotated(
-            session, direction, imageId, static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), bounds.x, bounds.y, 2,
-            height, offset.x, offset.y, height);
+            session, direction, imageId, { static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), height },
+            { bounds.x, bounds.y, 2 }, { offset.x, offset.y, height });
```
Comment (Gymnasiast) [inline src/openrct2/ride/transport/MiniatureRailway.cpp:1149]:
```suggestion
            { bounds, 2 }, { offset, height });
```

Comment (Gymnasiast) [reply src/openrct2/ride/transport/MiniatureRailway.cpp:1149]: Also: do we actually need the static cast to int8_t here?

File: src/openrct2/ride/transport/MiniatureRailway.cpp [hunk 3]
```diff
@@ -1253,8 +1253,8 @@ static void paint_miniature_railway_track_s_bend_right(
         imageId = miniature_railway_track_pieces_s_bend_right[direction & 1][trackSequence]
             | session.TrackColours[SCHEME_TRACK];
         PaintAddImageAsChildRotated(
-            session, direction, imageId, static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), bounds.x, bounds.y, 2,
-            height, offset.x, offset.y, height);
+            session, direction, imageId, { static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), height },
+            { bounds.x, bounds.y, 2 }, { offset.x, offset.y, height });
```
Comment (Gymnasiast) [inline src/openrct2/ride/transport/MiniatureRailway.cpp:1257]: See previous line note.

**Comment by duncanspumpkin** [comment]
I don't want to spend too much time adjusting the callers there are tons that aren't quite as efficient as can be but thats an exercise for a different day.

**Comment by Gymnasiast** [comment]
I get that, but why did you apply my requests _halfway_? There were only three of them anyway.

**Comment by duncanspumpkin** [comment]
> I get that, but why did you apply my requests _halfway_? There were only three of them anyway.

I've been thinking about this code for a while and I'm not convinced height should be part of these calls and making those changes makes it harder to understand

**Pull request merged**